packages/amplify-util-uibuilder/src/__tests__/handleEvents.test.ts

```diff
@@ -0,0 +1,27 @@
+import { run as runPostPull } from '../event-handlers/handle-PostPull';
+import { run as runPostPush } from '../event-handlers/handle-PostPush';
+import { run as generateComponents } from '../commands/generateComponents';
+
+jest.mock('../commands/generateComponents');
+
+describe('can handle events', () => {
+  let context: any;
+
+  beforeEach(() => {
+    context = {
+      amplify: {
+        invokePluginMethod: jest.fn(),
+      },
+    };
+  });
+
+  it('handles postPull', async () => {
+    await runPostPull(context);
+    expect(generateComponents).toHaveBeenCalledWith(context, 'PostPull');
```

Comment:
When creating the EventType, you can also make POST_PULL a constant instead of a string. This will allow for some nice help from IDE